src/OpenDiffix.Core/Aggregator.fs

```diff
@@ -382,6 +382,125 @@ type private DiffixSumNoise(summandType, aidsCount) =
       | Anonymizer.AnonymizedResult.NotEnoughAIDVs -> Null
       | Anonymizer.AnonymizedResult.Ok { NoiseSD = noiseSD } -> Real noiseSD
 
+
+type private CountHistogram(binSize: int64 option) =
+  let state = Dictionary<Value, int64>()
+
+  let generalize count =
+    match binSize with
+    | None
+    | Some 1L -> count
+    | Some binSize -> (float count / float binSize) |> floor |> int64 |> (*) binSize
+
+  interface IAggregator with
+    member this.Transition args =
+      match args with
+      | Null :: _ -> () // Ignore NULLs
+      | aid :: _ -> state |> Dictionary.increment aid
+      | _ -> invalidArgs args
+
+    member this.Merge aggregator =
+      let otherState = (castAggregator<CountHistogram> aggregator).State
+
+      for other in otherState do
+        let current = state |> Dictionary.getOrDefault other.Key 0L
+        state.[other.Key] <- current + other.Value
+
+    member this.Final(aggContext, anonContext) =
+      let bins = Dictionary<int64, int64>()
+
+      for pair in state do
+        let binLabel = generalize pair.Value
+        bins |> Dictionary.increment binLabel
+
+      bins
+      |> Seq.map (fun pair -> (pair.Key, pair.Value))
+      |> Seq.sortBy fst
+      |> Seq.map (fun (binLabel, aidCount) -> Value.List [ Integer binLabel; Integer aidCount ])
+      |> Seq.toList
+      |> Value.List
+
+  member this.State = state
+
+
+type private CountHistogramAidTracker = { mutable RowCount: int64; LowCount: DiffixLowCount }
+
+type private CountHistogramBin = { mutable AidCount: int64; LowCount: DiffixLowCount }
+
+type private DiffixCountHistogram(aidsCount, binSize: int64 option) =
+  let state = Dictionary<AidHash, CountHistogramAidTracker>()
```

Comment:
We can have an unhashed `Dictionary<Value, CountHistogramAidTracker>` here but this would lose consistency with a future postgres implementation.